content/automate/ManageIQ/System/CommonMethods/MiqAe.class/__methods__/weightedupdatestatus.rb

```diff
@@ -122,6 +136,11 @@ def main
                   # We set the task message.
                   if @handle.root['ae_state_step'] == 'on_error'
                     task.message = 'Failed'
+                    task.set_option('cleanup_request_id', create_cleanup_request.id)
+                  elsif task.get_option('cancel_requested')
+                    task.message = 'Cancelled'
+                    task.set_option('cleanup_request_id', create_cleanup_request.id)
+                    raise "Task cancellation requested."
```

Comment:
What is the result of raising an error? Will it eventually end with `Failed` and trigger another `create_cleanup_request`? 

Do we need to clear option `cancel_requested`? Will you check this flag multiple times and each time call `create_cleanup_request`?